Aftonbladet.js

```diff
@@ -0,0 +1,212 @@
+{
+	"translatorID": "235c5c47-0c04-4d77-9afe-e22265a017a9",
+	"label": "Aftonbladet",
+	"creator": "Jonatan Svensson Glad",
+	"target": "^https?://(ww(w|c)\\.)?aftonbladet\\.se",
+	"minVersion": "3.0",
+	"maxVersion": "",
+	"priority": 100,
+	"inRepository": true,
+	"translatorType": 4,
+	"browserSupport": "gcsibv",
+	"lastUpdated": "2016-08-10 13:45:00"
+}
+
+/*
+	Aftonbladet Translator - Parses Aftonbladet articles and creates Zotero-based metadata.
+	Copyright (C) 2016 Jonatan Svensson Glad
+	This program is free software: you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+	This program is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+	GNU General Public License for more details.
+	You should have received a copy of the GNU General Public License
+	along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+*/
+
+function detectWeb(doc, url) {
+	//TODO: adjust the logic here
+	if (url.indexOf('/article')>-1) {
+		return "newspaperArticle";
+	} else if (getSearchResults(doc, true)) {
+		return "multiple";
+	}
+}
+
+
+function getSearchResults(doc, checkOnly) {
+	var items = {};
+	var found = false;
+	//TODO: adjust the xpath
+	var rows = ZU.xpath(doc, '//a[contains(@href, "/article")]');
+	for (var i=0; i<rows.length; i++) {
+		//TODO: check and maybe adjust
+		var href = rows[i].href;
+		//TODO: check and maybe adjust
+		var title = ZU.xpathText(rows[i], './/h2');
+		if (!href || !title) continue;
+		if (checkOnly) return true;
+		found = true;
+		items[href] = title;
+	}
+	return found ? items : false;
+}
+
+
+function doWeb(doc, url) {
+	if (detectWeb(doc, url) == "multiple") {
+		Zotero.selectItems(getSearchResults(doc, false), function (items) {
+			if (!items) {
+				return true;
+			}
+			var articles = new Array();
+			for (var i in items) {
+				articles.push(i);
+			}
+			ZU.processDocuments(articles, scrape);
+		});
+	} else {
+		scrape(doc, url);
+	}
+}
+
+function scrape(doc) {
+	var newArticle = new Zotero.Item('newspaperArticle');
+		newArticle.title = ZU.xpathText(doc, "//h1");	
+		newArticle.date = ZU.xpathText(doc, '//time[@pubdate]');
+		var author = ZU.xpathText(doc, '//address[@class="abByline"]');
```

Comment:
Try something like this (I haven't tested it):
```javascript
var authors = ZU.xpath(doc, '//address[@class="abByline"]/div[contains(@class, "abAuthor")]');
for (var i=0; i<authors.length; i++) {
   newArticle.creators.push(ZU.cleanAuthor(authors[i].textContent, "author"));
}
```
BTW this should also solve the other problem with paywalled sites (and no author info given on the free part).